### ISSUE TITLE
Replace npm package "decompress"

### ISSUE THREAD

**Issue by tylerforesthauser** [gridsome/gridsome#1012]
## Summary

Replace npm package `decompress` due to vulnerability documented [here](https://www.npmjs.com/advisories/1217).

### Motivation

> All versions of decompress are vulnerable to Arbitrary File Write. The package fails to prevent extraction of files with relative paths, allowing attackers to write to any folder in the system by including filenames containing../.

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-mozjpeg > mozjpeg > bin-build >          │
│               │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-pngquant > pngquant-bin > bin-build >    │
│               │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-webp > cwebp-bin > bin-build >           │
│               │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-mozjpeg > mozjpeg > bin-build > download │
│               │ > decompress                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-pngquant > pngquant-bin > bin-build >    │
│               │ download > decompress                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-webp > cwebp-bin > bin-build > download  │
│               │ > decompress                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-mozjpeg > mozjpeg > bin-wrapper >        │
│               │ download > decompress                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-pngquant > pngquant-bin > bin-wrapper >  │
│               │ download > decompress                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Write                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ decompress                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gridsome                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gridsome > imagemin-webp > cwebp-bin > bin-wrapper >         │
│               │ download > decompress                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1217                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by noxify** [comment]
Related issue: https://github.com/kevva/decompress/issues/71

**Comment by hjvedvik** [comment]
I see there is an open PR for `decompress` which hopefully will be merged soon.

**Comment by noxify** [comment]
@hjvedvik it seems that the maintainer of the decompress package is not active currently.

Based on the comments and other repos ( like gatsby https://github.com/gatsbyjs/gatsby/pull/22014#issuecomment-596636050 ) it's a "no real issue" for gridsome. ( but please correct me, if I'm wrong ) 

Maybe we could update the readme and mark this as known issue?

**Comment by a-kriya** [comment]
Fixed in release v4.2.1: https://github.com/kevva/decompress/releases

**Comment by ChristianSantos07** [comment]
abra o cmd e rode 

npm install npm@latest -g

ela pega a ultima versão
